050-Kafka-Node-Pools.md

```diff
@@ -0,0 +1,832 @@
+This is however not part of this proposal.
+Until this is implemented, the partition replicas can be moved also using the `kafka-reassign-partitions.sh` tool.
+
+Despite the risks and limitations, having the ability to be able to move the nodes between node pools is still desired and might be useful.
+So it should still be supported, even if it might not be part of the initial implementation.
+
+### Deletion of the `KafkaNodePool` resource
+
+When the `KafkaNodePool` resource belonging to an existing Kafka cluster is deleted, all the resources belonging to it and managed by it will be deleted as well.
+This will also include the Pods and other resources.
+The PVCs will be deleted only if the `deleteClaim` flag was set to `true` in the storage configuration (default value for `deleteClaim` is `false` - so it will not be deleted by default).
+While it is possible that the user deleted the `KafkaNodePool` by mistake, once the resource is deleted, there is not much the operator can do about it.
+This behavior does not differ from what happens if the `Kafka` custom resource is deleted, so it does not introduce any new behavior.
+Users can use other mechanisms such as RBAC or finalizers to prevent deleting the custom resources by mistake.
+
+### Implementation
+
+#### Feature Gate
+
+The node pools will be introduced gradually using a Feature Gate.
+The new Feature gate will be named `KafkaNodePools`.
+The following table shows the expected graduation of the `KafkaNodePools` feature gate:
+
+| Phase | Strimzi versions       | Default state                                          |
+|:------|:-----------------------|:-------------------------------------------------------|
+| Alpha | 0.35 - 0.37            | Disabled by default                                    |
+| Beta  | 0.38 - 0.39            | Enabled by default                                     |
+| GA    | 0.40 and newer         | Enabled by default (without possibility to disable it) |
+
+The main purpose of the feature gate is in this case to protect users from a feature which is not mature enough and might be removed in future versions.
+Only small parts of the code will actually depend on the feature gate being enabled or disabled.
+
+The annotation `strimzi.io/node-pools: enabled` on the `Kafka` custom resource will be also required in order to use the `KafkaNodePool` resources on a given Kafka cluster.
+
+Since the KRaft mode will be supported only with `KafkaNodePool` resources, the `UseKRaft` feature gate will be configured as dependent on the `KafkaNodePools` feature gate.
+It will not be possible to enable `UseKRaft` feature gate without also having `KafkaNodePools` feature gate enabled.
+
+#### Virtual node pool
+
+To avoid two parallel implementations in the Strimzi source code, the operator will use _virtual node pool_ when the feature gate is disabled.
```

Comment:
This will be a virtual node pool named `kafka` right?

Reply:
Yes, it will be named `kafka`. That way, all the resources will have the same names as today.